### PR TITLE
Image storage location

### PR DESCRIPTION
**Pull request by shubhamkr619**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/6082

This is the PR to support StorageLocation for Image resource. This shall allow users to select the region and supply that as a list. 
However, this PR could not be tested because of the error : https://github.com/hashicorp/terraform-provider-google/issues/13846

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes). Reference issue: https://github.com/hashicorp/terraform-provider-google/issues/6082
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
* Added support for storageLocation in Image resource 
```

### CODE REVIEW

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -6302,6 +6302,13 @@ objects:
         description: |
           An optional description of this resource. Provide this property when
           you create the resource.
+      - !ruby/object:Api::Type::Array
+        name: 'storageLocations'
+        description: |
+          Cloud Storage bucket storage location of the image 
+          (regional or multi-regional). 
+          Reference link: https://cloud.google.com/compute/docs/reference/rest/v1/images
+        item_type: Api::Type::String
```
Comment (ScottSuarez) [inline mmv1/products/compute/api.yaml:6311]: Looks like this value needs to be set to computed as the api is returning defaults for this field.

Comment (shubhamkr619) [reply mmv1/products/compute/api.yaml:6311]: let me check this

Comment (shubhamkr619) [reply mmv1/products/compute/api.yaml:6311]: @ScottSuarez can you give me an example of a property where the computed value is userd?

File: mmv1/templates/terraform/examples/image_basic_storage_location.tf.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+resource "google_compute_image" "example" {
+  name = "<%= ctx[:vars]['image_name'] %>"
+
+  raw_disk {
+    source = "https://storage.googleapis.com/bosh-gce-raw-stemcells/bosh-stemcell-97.98-google-kvm-ubuntu-xenial-go_agent-raw-1557960142.tar.gz"
```
Comment (ScottSuarez) [inline mmv1/templates/terraform/examples/image_basic_storage_location.tf.erb:5]: This test seems to rely on an external disk image in a personal project. Is that so?

Comment (ScottSuarez) [reply mmv1/templates/terraform/examples/image_basic_storage_location.tf.erb:5]: Ahh nevermind.. This value is already used in other tests, apologies

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/products/compute/api.yaml:6311]: Let me get back to you `default_from_api` should traditionally cover this. Not sure what's going on!

Comment (ScottSuarez) [reply mmv1/products/compute/api.yaml:6311]: ohh I see you just added it in a recent change. Let me rerun tests for you then. Should be covered

**Comment by ScottSuarez** [comment]
Ah, I think this will need another rebase.

Seems the engine team refactored some attributes. I'm not sure why the CI is dying. So sorry, for any delays here!!
https://github.com/GoogleCloudPlatform/magic-modules/pull/7431/files

**Comment by shubhamkr619** [comment]
@ScottSuarez , i have rebased it. Please validate.

**Comment by shubhamkr619** [comment]
@ScottSuarez , it seems like the cla issue persist. What do you recommend here?

**Comment by ScottSuarez** [comment]
/gcbrun
let me try to turn off the cla

**Comment by ScottSuarez** [comment]
Yeah looks like the builds keep dying due to the cla requirements. Could you click details on the CLA failure below the PR and try to sign it. It's strange because you are a googler so I don't know why it's failing atm.

**Comment by ScottSuarez** [comment]
Looks like there was a bad rebase too, 1420 files are changed according to github. Feel free to reach out of your need any help.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
waiting for rebase.. feel free to rerequest a review when you need

**Pull request closed without merging**